app/services/cart_service.rb

```diff
@@ -1,54 +1,37 @@
 require 'open_food_network/scope_variant_to_hub'
 
-Spree::OrderPopulator.class_eval do
+class CartService
+  attr_accessor :order, :currency
   attr_reader :variants_h
+  attr_reader :errors
+
+  def initialize(order, currency)
+    @order = order
+    @currency = currency
```

Comment:
Does the currency need to be a parameter here, or can you just have `order` and do `@currency = order.currency`?

Reply:
I have just copied this code from the existing orders_controller and improved the easy stuff. This one was a bit longer to do. But now that you mention it, it's done :-)

I added a new commit to the PR with that change. I dont think it will create any issues, order.currency is being populated from current_currency as well.